### PR TITLE
Feature/get involved page

### PR DESCRIPTION
**Pull request by croab**
## Status

* Current Status: Ready for review
* Review App link(s): https://teachcomputing-pr-2054.herokuapp.com/
* Closes [#2694](https://github.com/NCCE/teachcomputing.org-issues/issues/2694)

## Review progress:

- [x] Browser tested
- [ ] Front-end review completed
- [ ] Tech review completed

## What's changed?

- Get Involved page has been replaced with new designs
- The following pages have been removed and redirects to the get involved page put in place:
-- https://teachcomputing.org/supporting-partners
-- https://teachcomputing.org/contributing-partners

## Outstanding

- To check whether it is indeed OK to exclude iframes from system tests

### CODE REVIEW

File: app/views/pages/get-involved.html.erb [hunk 1]
```diff
@@ -1,55 +1,183 @@
...
+      <div class="govuk-grid-row">
+        <div class="govuk-grid-column-three-quarters ncce-about__content ncce-section">
+          <p class="govuk-body-m"><%= t(".intro.body_1") %></p>
+          <p class="govuk-body-s govuk-!-margin-bottom-0"><%= t(".intro.body_2") %></p>
+        </div>
+
+        <div class="govuk-grid-column-one-quarter ncce-enrichment-card govuk-!-padding-4 ncce-section">
+          <h2 class="govuk-heading-s icon-chart"><%= t(".intro.box.heading") %></h2>
+          <p class="govuk-body-s"><%= t(".intro.box.body") %></p>
+        </div>
```
Comment (GDTimi) [inline app/views/pages/get-involved.html.erb:15]: Although the icon chart is referenced in the class description, it isn't rendering for me locally?

Screenshot 2024-06-05 at 15 01 14

File: app/views/pages/get-involved.html.erb [hunk 2]
```diff
@@ -1,55 +1,183 @@
...
-      <div class="govuk-grid-column-full govuk-!-margin-top-3">
-        <%= render BorderedCardsComponent.new(**(GetInvolved.partnership_cards)) %>
+  </div>
+
+  <%= render BannerWithBackgroundComponent.new(
+    background_image_css: "get-involved-background_classroom",
+    label: {
+      title: t(".impact.heading"),
+      body: t(".impact.body"),
+    }
```
Comment (GDTimi) [inline app/views/pages/get-involved.html.erb:94]: It looks like the star file is available (.icon-star-black / star-black.svg), but don't see it referenced for display?

File: spec/views/pages/get-involved.html_spec.rb [hunk 3]
```diff
@@ -1,23 +1,59 @@
...
+  it "has a video for both ambassador and governor sections" do
+    expect(rendered).to have_css(".banner-component__media-section", count: 2)
+  end
+
+  it "has a get in touch section" do
+    expect(rendered).to have_css(".govuk-heading-m", text: "Deliver social value through investing in improving young people’s life outcomes")
+    expect(rendered).to have_link("Get in touch", href: governors_and_trustees_url)
   end
 
   it "has a get involved section" do
```
Comment (GDTimi) [inline spec/views/pages/get-involved.html_spec.rb:39]: Perhaps rename to "making an impact today section"?

File: app/controllers/pages_controller.rb [hunk 4]
```diff
@@ -2,7 +2,14 @@ class PagesController < ApplicationController
...
   before_action :redirect_to_dashboard, only: [:login]
 
+  RETIRED_SLUGS = {
+    "supporting-partners" => :get_involved_path,
+    "contributing-partners" => :get_involved_path
+  }.freeze
+
   def page
+    redirect_to send(RETIRED_SLUGS[params[:page_slug]]) and return if RETIRED_SLUGS.key?(params[:page_slug])
+
```
Comment (GDTimi) [inline app/controllers/pages_controller.rb:12]: This is a bit interesting, as I think there must have been retirements and redirects done previously? This works, I just wonder if there are any previous solutions / ways of dealing with these retire and redirect scenarios.

**Review by GDTimi — CHANGES_REQUESTED** [review]
Great work, a lot done here! I've added a couple of minor comments, main issue seems to be not seeing icons I expected to (let me know if this is potentially something on my end!). 👍

File: app/views/pages/get-involved.html.erb (hunk 1 above)
Comment (croab) [reply app/views/pages/get-involved.html.erb:15]: @GDTimi - might need to restart your docker containers but if that doesn't work then give me a shout!

File: app/views/pages/get-involved.html.erb (hunk 2 above)
Comment (croab) [reply app/views/pages/get-involved.html.erb:94]: Perhaps ditto to the above

File: spec/views/pages/get-involved.html_spec.rb (hunk 3 above)
Comment (croab) [reply spec/views/pages/get-involved.html_spec.rb:39]: The making an impact section is the one with the background below this I believe, so will probably leave as is for this

File: app/controllers/pages_controller.rb (hunk 4 above)
Comment (croab) [reply app/controllers/pages_controller.rb]: Thanks for this! I moved the redirects into the routes file

**Review by msquance-stem — APPROVED** [review]

File: app/views/pages/get-involved.html.erb (hunk 2 above)
Comment (GDTimi) [reply app/views/pages/get-involved.html.erb]: Needed a server restart on my side, all good 👍

File: app/views/pages/get-involved.html.erb (hunk 1 above)
Comment (GDTimi) [reply app/views/pages/get-involved.html.erb:16]: Needed a server restart on my side, all good 👍

**Review by GDTimi — APPROVED** [review]

**Pull request merged**